Agents/PropertyValueEstimationAgent/tests/conftest.py

```diff
@@ -0,0 +1,380 @@
+# Session-scoped test fixtures
+# (i.e. the fixture is destroyed at the end of the test session)
+# ----------------------------------------------------------------------------------
+
+@pytest.fixture(scope="session")
+def get_blazegraph_service_url(session_scoped_container_getter):
+    def _get_service_url(service_name, url_route):
+        service = session_scoped_container_getter.get(service_name).network_info[0]
+        service_url = f"http://localhost:{service.host_port}/{url_route}"
+
+        # This will run only once per entire test session
+        # It ensures that the requested Blazegraph Docker service is ready to accept SPARQL query/update
+        service_available = False
+        while not service_available:
+            try:
+                response = requests.head(service_url)
+                if response.status_code != requests.status_codes.codes.not_found:
+                    service_available = True
+            except requests.exceptions.ConnectionError:
+                time.sleep(3)
+
+        return service_url
+    return _get_service_url
+
+
+@pytest.fixture(scope="session")
+def get_postgres_service_url(session_scoped_container_getter):
+    def _get_service_url(service_name, url_route):
+        service = session_scoped_container_getter.get(service_name).network_info[0]
+        service_url = f"jdbc:postgresql://localhost:{service.host_port}/{url_route}"
+
+        # This will run only once per entire test session
+        # It ensures that the requested PostgreSQL Docker service is ready to accept queries
+        service_available = False
+        while not service_available:
+            try:
+                conn = pg.connect(host='localhost', port=service.host_port,
+                                  user=DB_USER, password=DB_PASSWORD,
+                                  database=DATABASE)
+                if conn.status == pg.extensions.STATUS_READY:
+                    service_available = True
+            except Exception:
+                time.sleep(3)
+
+        return service_url
+    return _get_service_url
+
+
+@pytest.fixture(scope="session")
+def initialise_clients(get_blazegraph_service_url, get_postgres_service_url):
+    # Retrieve "user-facing" endpoints for all clients/services relevant for testing
+    # (i.e. invoked during testing from outside the Docker stack)
+    # --> those shall be `localhost:...` even when agent is running as Docker container
+    
+    # Retrieve endpoint for triple store
+    sparql_endpoint = get_blazegraph_service_url(KG_SERVICE, url_route=KG_ROUTE)
+
+    # Retrieve endpoint for postgres
+    rdb_url = get_postgres_service_url(RDB_SERVICE, url_route=RDB_ROUTE)
+
+    # Create SparqlClient for testing
+    sparql_client = KGClient(sparql_endpoint, sparql_endpoint)
+
+    # Create DerivationClient for creating derivation instances
+    derivation_client = sparql_client.jpsBaseLib_view.DerivationClient(
```

Comment:
It's fine to create `derivation_client` directly from Java side here as we didn't do pytest with type hints, but in case you want to use the derivation client in your input agents later, I would then recommend instantiating `pyderivationagent.PyDerivationClient` which includes type hints so easier to use.

Reply:
Addressed in [1447df6](https://github.com/cambridge-cares/TheWorldAvatar/pull/454/commits/1447df66cc776b90a98c00c2abfd1fe42889e786)